### PR TITLE
Feature/django

### PR DESCRIPTION
**Pull request by tijmenbaarda**
Conversion of sync scripts to a Django setup.

I have not extensively tested if this behaves as expected when changes or additions are made, but it seems to work correctly.

### CODE REVIEW

**Review by jgonggrijp — COMMENTED** [review]
> (...) I'd like to ask for @jgonggrijp's opinion on a few things:
> 
>  1. Should we make the synchronization step only store the `annotation.content` and `publication.content` API response data (and required keys), and do all processing in another step?

I don't have a strong opinion on the matter; the current version of the code does only minimal processing, so at least in terms of performance, I think it will be harmless. That being said, I don't know why @tijmenbaarda chose to extract the publication's `attachment_id` and the annotation's `comment` already at this time. If that solves a problem, by all means keep it like that, if it doesn't, maybe consider moving it to the later processing step just to keep related things in one place.

> 
>  2. Does it make sense to use separate Annotation and Publication models just for synchronizing the raw API responses, and create other models that have the actual fields we'll want to visualize, search, and filter on? (...)

Yes. But maybe this is exactly why Tijmen placed these import-related models within the `sync` app? I can imagine that he was planning to create another app for the fully processed version, which would have its own `Annotation` and `Publication` models, as well as a whole bunch of other, more fine-grained ones. Django scopes models per app when it creates the corresponding tables in the underlying database.

> 
>  3. Any suggestions for the code in `zoterosync.py` specifically,

Yes, just a few organizational things. I put them in comments below.

> or for the Django setup in general?

No, it looks very standard to me.

**Comment by ar-jan** [comment]
> That being said, I don't know why @tijmenbaarda chose to extract the publication's attachment_id and the annotation's comment already at this time.

That was a left-over from my initial pure SQLite test, where it was needed to link annotations to publications, but we'll now do that elsewhere with the rest of the processing we still have to add.

File: lidiabrowser/sync/zoterosync.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    items = zot.everything(zot.top(since=since))
+    if items:
+        for item in items:
+            key = item["key"]
+            obj = item
+            # Annotations are linked to their parent PDF, not the bibliographic
+            # item
+            # Assuming only one attachment per publication
+            url: str = obj.get("links", {}).get("attachment", {}).get("href")
+            attachment_id: str = url.rstrip("/").split("/")[-1] if url else ""
```
[the remark this answers is not in the document]
Comment (tijmenbaarda) [reply lidiabrowser/sync/zoterosync.py:37]: Removed this

File: lidiabrowser/sync/zoterosync.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    library version.
+    Saves the annotations to database.
+    """
+    # TODO: use Zotero.follow() or iterfollow() methods
+    # https://pyzotero.readthedocs.io/en/latest/#the-follow-and-everything-methods
+    items = zot.everything(zot.items(itemType="annotation", since=since))
+    if items:
+        for item in items:
+            key = item["key"]
+            comment = item["data"]["annotationComment"]
```
[the remark this answers is not in the document]
Comment (tijmenbaarda) [reply lidiabrowser/sync/zoterosync.py:58]: Removed this too

File: lidiabrowser/sync/models.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+class Publication(models.Model):
+    zotero_id = models.CharField(max_length=100, unique=True)
+    content = models.JSONField(default=dict)
+    attachment_id = models.CharField(max_length=100)
+
+
+class Annotation(models.Model):
+    zotero_id = models.CharField(max_length=100, unique=True)
+    content = models.JSONField(default=dict)
+    comment = models.JSONField(default=dict)
```
[the remark this answers is not in the document]
Comment (tijmenbaarda) [reply lidiabrowser/sync/models.py:13]: I have removed the `comment` field.

File: lidiabrowser/sync/zoterosync.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tijmenbaarda) [reply lidiabrowser/sync/zoterosync.py:37]: I have removed these lines completely.

**Comment by tijmenbaarda** [comment]
Merging this now, but we need to find a way to properly test if this works as it should. I will make an issue.

**Pull request merged**